client/packages/android/app/src/main/java/org/openmsupply/client/NativeApi.java

```diff
@@ -365,8 +365,6 @@ private String getHostAddress(NsdServiceInfo serviceInfo, Boolean isLocal){
             return serviceInfo.getHost().getHostAddress();
         } catch (Exception ex) {
             Log.e(OM_SUPPLY, ex.toString());
-        }
-        finally {
```

Comment:
Oops sorry that was my suggestion.

But I can't quite see the difference here sorry. 

`return inetAddress.getHostAddress();` if it's local and we can get local address without error then return
`return serviceInfo.getHost().getHostAddress();` if not local and no addresses found that match the criteria return .getHostAddress()

Otherwise do finally return "127.0.0.1";

In the original code, before my suggestion, it was slightly different, if error happens while getting isLocal address, we ignore it silently and try  `return serviceInfo.getHost().getHostAddress();`, if that fails, only then we return `127.0.0.1`, sorry for the confusion, maybe we should revert back to the original (if when isLocal there is an error and we want to try returning `return serviceInfo.getHost().getHostAddress();` rather then `127.0.0.1`

Original, [as per this diff](https://github.com/msupply-foundation/open-msupply/pull/2527/commits/0212fe8e34d4320fa6ec8159968cfb5010bd67e3#diff-b188c1ef5b0d6a28061281e09a8530f0165465d438ea1a4e16e76390658c3d06)

```java
     if (isLocal) {
            try {
                for (Enumeration<NetworkInterface> en = NetworkInterface.getNetworkInterfaces(); en.hasMoreElements(); ) {
                    NetworkInterface intf = en.nextElement();
                    for (Enumeration<InetAddress> enumIpAddr = intf.getInetAddresses(); enumIpAddr.hasMoreElements(); ) {
                        InetAddress inetAddress = enumIpAddr.nextElement();
                        if (!inetAddress.isLoopbackAddress() && !inetAddress.isLinkLocalAddress() && inetAddress.isSiteLocalAddress()) {
                            return inetAddress.getHostAddress();
                        }
                    }
                }
            } catch (Exception ex) {
                Log.e(OM_SUPPLY, ex.toString());
            }
        }
        try {
            return serviceInfo.getHost().getHostAddress();
        } catch (Exception ex) {
            Log.e(OM_SUPPLY, ex.toString());
        }
        // with no network available the getHostAddress() will
        // throw an Exception - in this case default to local loopback
        // as no other network is reachable!
        return "127.0.0.1";
```

Reply:
I'm reluctant to refactor again, this one takes a lot of time to test. All the same, I've simplified - with an early return if the IP isn't local, since that's the whole point of the original change.

Now the logic is
- is host remote? return `serviceInfo.getHost().getHostAddress()` ( the original method )
- if it's local and we can get local address without error then return `inetAddress.getHostAddress()`
- if an error happened, log the error and return `serviceInfo.getHost().getHostAddress()`

and don't try to return `127.0.0.1` that was only introduced because `getHostAddress` can throw an exception and Android Studio was unhappy if that is not caught by something.

How does that sound? I've retested this approach and it resolves the local IP correctly and fixes the 'all the servers are 127.0.01 issue'

Reply:
Thank you Mark, and sorry if my previous suggestion cause more hassle